### PR TITLE
Do not handle universal quantification on functions in model-based FMF

### LINKED ISSUE #4225: Segfault finite-model-find
```
[543] % cvc4 -q small.smt2
CVC4 suffered a segfault.
Offending address is 0x40
Looks like a NULL pointer was dereferenced.
Aborted
[544] % 
[544] % cat small.smt2
(set-option :finite-model-find true)
(assert (forall ((a (-> Int Int)) (b Int)) (not (= (a b) 0))))
(check-sat)
[545] %
```

OS: Ubuntu 18.04
Commit: 82f5610

### LINKED ISSUE #4289: Segfault sygus-inference, elim-ext-arith-quant, finite-model-find
```
[542] % cvc4-1.7 -q small.smt2
sat
[543] % cvc4 -q small.smt2
CVC4 suffered a segfault.
Offending address is 0x40
Looks like a NULL pointer was dereferenced.
Aborted
[544] % 
[544] % cat small.smt2
(set-option :sygus-inference true)
(set-option :elim-ext-arith-quant true)
(set-option :finite-model-find true)
(declare-fun a () Real)
(assert (is_int (/ 0 a)))
(check-sat)
[545] %
```

OS: Ubuntu 18.04
Commit: 0073464

### LINKED ISSUE #4483: (fmf-fun) Segmentation fault on UFLIA formula
Hi,
For this case, CVC4  throws out a segmentation fault:
```
[567] % cvc4 small.smt2
unknown
[568] % cvc4 --fmf-fun small.smt2
CVC4 suffered a segfault.
Offending address is 0x40
Looks like a NULL pointer was dereferenced.
Aborted
[569] % 
[569] % cat small.smt2
(set-logic ALL)
(declare-fun a (Int) Bool)
(assert (forall ((b (-> Int Int)) (c Int)) (a (b c))))
(check-sat)
[570] %
```

OS: Ubuntu 18.04
Commit: 00badd3

### PR DESCRIPTION
**Pull request by ajreynol**
Fixes #4225, fixes https://github.com/CVC4/cvc4-projects/issues/159, fixes https://github.com/CVC4/cvc4-projects/issues/157, fixes #4289, fixes #4483.

This makes it so that the main model-based instantiation algorithm is not applied to quantified formulas with universally quantified functions.

Identation changed in a FMF function, this was refactored to conform to guidelines, and further cleaned.

### CODE REVIEW

File: test/regress/regress1/fmf/issue4225-univ-fun.smt2 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+; COMMAND-LINE: --finite-model-find
```
Comment (HanielB) [inline test/regress/regress1/fmf/issue4225-univ-fun.smt2:1]: shouldn't this benchmark be using `--uf-ho`?

File: src/theory/quantifiers/fmf/full_model_check.cpp [hunk 2]
```diff
@@ -1290,3 +1319,33 @@ Node FullModelChecker::getFunctionValue(FirstOrderModelFmc * fm, Node op, const
...
+  }
+  TypeNode typ = nm->mkFunctionType(types, nm->booleanType());
+  Node op = nm->mkSkolem("qfmc", typ, "op for full-model checking");
+  d_quant_cond[q] = op;
+}
+
+bool FullModelChecker::isHandled(Node q) const
+{
+  return d_unhandledQuant.find(q) == d_unhandledQuant.end();
+}
```
Comment (HanielB) [inline src/theory/quantifiers/fmf/full_model_check.cpp:1351]: Shouldn't this function also check that the quantifier was registered?

File: test/regress/regress1/fmf/issue4225-univ-fun.smt2 (hunk 1 above)
Comment (ajreynol) [reply test/regress/regress1/fmf/issue4225-univ-fun.smt2:1]: Yes I'll add.

File: src/theory/quantifiers/fmf/full_model_check.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/quantifiers/fmf/full_model_check.cpp:1351]: It's assumed that the quantified formula q is registered here.

**Pull request merged**